hack/update_owners.py

```diff
@@ -0,0 +1,111 @@
+        content = zlib.decompress(content, 15 | 16)
+    return json.loads(content)
+
+
+def normalize(name):
+    name = re.sub(r'\[.*?\]|\{.*?\}', '', name)
+    name = re.sub(r'\s+', ' ', name)
+    return name.strip()
+
+
+def load_owners(fname):
+    owners = {}
+    for n, (name, owner, random_assignment) in enumerate(csv.reader(open(fname))):
+        if n == 0:
+            continue  # header
+        owners[normalize(name)] = (owner, int(random_assignment))
+    return owners
+
+
+def write_owners(fname, owners):
+    out = csv.writer(open(fname, 'w'))
+    out.writerow(['name', 'owner', 'auto-assigned'])
+    for name, (owner, random_assignment) in sorted(owners.items()):
+        out.writerow([name, owner, int(random_assignment)])
+
+
+def get_maintainers():
+    # Github doesn't seem to support team membership listing without org admin privileges.
+    # Instead, we do it manually:
+    # Open https://github.com/orgs/kubernetes/teams/kubernetes-maintainers
+    # Run this in the js console:
+    # [].slice.call(document.querySelectorAll('.team-member-username a')).map(e => e.textContent.trim())
+    return ["a-robinson", "alex-mohr", "amygdala", "andyzheng0831", "apelisse", "aronchick", 
+        "ArtfulCoder", "bgrant0607", "bgrant0607-nocc", "bprashanth", "brendandburns", 
+        "caesarxuchao", "childsb", "cjcullen", "david-mcmahon", "davidopp", "dchen1107", 
+        "deads2k", "derekwaynecarr", "dubstack", "eparis", "erictune", "fabioy", 
+        "fejta", "fgrzadkowski", "freehan", "ghodss", "girishkalele", "gmarek", 
+        "goltermann", "grodrigues3", "hurf", "ingvagabund", "ixdy", 
+        "jackgr", "janetkuo", "jbeda", "jdef", "jingxu97", "jlowdermilk", 
+        "jsafrane", "jszczepkowski", "justinsb", "kargakis", "karlkfi", 
+        "kelseyhightower", "kevin-wangzefeng", "krousey", "lavalamp", 
+        "liggitt", "luxas", "madhusudancs", "maisem", "mansoorj", 
+        "matchstick", "mikedanese", "mml", "mtaufen", "mwielgus", "ncdc", 
+        "nikhiljindal", "piosz", "pmorie", "pwittrock", "Q-Lee", "quinton-hoole", 
+        "Random-Liu", "rmmh", "roberthbailey", "ronnielai", "saad-ali", "sarahnovotny", 
+        "smarterclayton", "soltysh", "spxtr", "sttts", "swagiaal", "thockin", 
+        "timothysc", "timstclair", "tmrts", "vishh", "vulpecula", "wojtek-t", "xiang90", 
+        "yifan-gu", "yujuhong", "zmerlynn"]
+
+
+def main():
+    test_history = get_test_history()
+    test_names = sorted(set(map(normalize, test_history['test_names'])))
+    owners = load_owners(OWNERS_PATH)
+
+    outdated_tests = sorted(set(owners) - set(test_names))
+    new_tests = sorted(set(test_names) - set(owners))
+    maintainers = get_maintainers()
+
+    print '# OUTDATED TESTS (%d):' % len(outdated_tests)
+    print  '\n'.join(outdated_tests)
+    print '# NEW TESTS (%d):' % len(new_tests)
+    print  '\n'.join(new_tests)
+
+    for name in outdated_tests:
+        owners.pop(name)
```

Comment:
Interesting so this prints info about new and outdated tests as well as random assignments who are no longer maintainers, and it removes outdated tests but otherwise just prints warnings?